prover/src/channel.rs

```diff
@@ -86,10 +86,20 @@ where
     /// Saves the evaluations of trace polynomials over the out-of-domain evaluation frame. This
     /// also reseeds the public coin with the hashes of the evaluation frame states.
     pub fn send_ood_trace_states(&mut self, trace_states: &[Vec<E>]) {
-        self.ood_frame.set_trace_states(trace_states);
-        for trace_state in trace_states {
-            self.public_coin.reseed(H::hash_elements(trace_state));
+        debug_assert_eq!(trace_states.len(), FRAME_SIZE);
+
+        // save the evaluations with the current and next evaluations interleaved for each polynomial
+        let width = trace_states[0].len();
+        let mut result = vec![E::ZERO; width * FRAME_SIZE];
+        for i in 0..width {
+            for j in 0..FRAME_SIZE {
+                result[i * FRAME_SIZE + j] = trace_states[j][i];
+            }
         }
```

Comment:
I'd probably do this as something like:
```Rust
let size = trace_states.len();
let width = trace_states[0].len();
let mut result = Vec::new();
for i in 0..width {
    for j in 0..size {
        result.push(trace_states[j][i]);
    }
}
```
This way we don't need to tie the code to `FRAME_SIZE` constant.

Reply:
Done!